includes/class-dispatcher.php

```diff
@@ -87,6 +88,48 @@ public function _execute_view( $template ) {
 		return $template;
 	}
 
+    /**
+     * Redirect from draft listings to the main directory page.
+     *
+     * @since x.x.x
+     *
+     * @return void
+     */
+    public function maybe_redirect_from_draft_listings() {
+        global $wp_query, $post, $wpdb;
+
+        if ( $post !== NULL ) {
```

Comment:
Could we return immediately if we only run this if is_404()? I'm not sure if it would return true at this point or not, but I think it might.

Reply:
I'll address the question above here because I can't comment on that one.

>Would ! $post do the job here?

It would but without the negation 😄 . I'll change it to that, thanks!

>Could we return immediately if we only run this if is_404()? I'm not sure if it would return true at this point or not, but I think it might.

Good idea! That would remove the need for the $post global!